src/main/java/com/faforever/client/leaderboard/LeaderboardService.java

```diff
@@ -172,11 +171,24 @@ public CompletableFuture<Optional<LeagueEntryBean>> getHighestActiveLeagueEntryF
             .max((e1, e2) -> SUBDIVISION_COMPARATOR.compare(e1.getSubdivision(), e2.getSubdivision())));
   }
 
-  public CompletableFuture<Optional<LeagueEntryBean>> getActiveLeagueEntryForPlayer(PlayerBean player, LeaderboardBean leaderboard) {
-    return getActiveLeagueEntriesForPlayer(player)
-        .thenApply(leagueEntries -> leagueEntries.stream()
-            .filter(leagueEntry -> Objects.equals(leagueEntry.getLeagueSeason().getLeaderboard(), leaderboard))
-            .findFirst());
+  @Cacheable(value = CacheNames.LEAGUE_ENTRIES, sync = true)
+  public CompletableFuture<Optional<LeagueEntryBean>> getActiveLeagueEntryForPlayer(PlayerBean player, String leaderboardName) {
+    ElideNavigatorOnCollection<LeagueSeasonScore> navigator = ElideNavigator.of(LeagueSeasonScore.class).collection()
+        .setFilter(qBuilder()
+            .intNum("loginId").eq(player.getId())
+            .and()
+            .string("leagueSeason.leaderboard.technicalName").eq(leaderboardName)
+            .and()
+            .instant("leagueSeason.startDate").before(OffsetDateTime.now().toInstant(), false)
+            .and()
+            .instant("leagueSeason.endDate").after(OffsetDateTime.now().toInstant(), false)
+        );
+    return fafApiAccessor.getMany(navigator)
+        .next()
+        .map(dto -> leaderboardMapper.map(dto, player, new CycleAvoidingMappingContext()))
+        .filter(leagueEntryBean -> leagueEntryBean.getSubdivision() != null)
```

Comment:
Couldn't this be made into an isnull check on the filter

Reply:
Do you mean like this? `.filter(leagueEntryBean -> Objects.nonNull(leagueEntryBean.getSubdivision()))`
Not sure if this really improves readability

Reply:
No you are filtering java side, when you could just add the bull check to the API call

Reply:
`.and().string("leagueSeasonDivisionSubdivision").exists()` doesn't work. It seems to break the api call.